### ISSUE TITLE
JSON Parse Error

### ISSUE THREAD

**Issue by DarioS** [BioinformaticsFMRP/TCGAbiolinks#115]
It fails at the last step.

```
RNAquery <- GDCquery(project = "TCGA-SKCM",  data.category = "Transcriptome Profiling",
                     data.type = "Gene Expression Quantification",
                     workflow.type = "HTSeq - Counts")
GDCdownload(RNAquery, chunks.per.download = 4)
RNAcounts <- GDCprepare(RNAquery)
```

The error is in the header.

```
Starting to add information to samples
 => Add clinical information to samples
Error: lexical error: invalid char in json text.
                                       <?xml version="1.0" ?> <respons
                     (right here) ------^
```

I have `jsonlite` version 1.5 and `rjson` version 0.2.15 loaded.

**Comment by AswathyPaul** [comment]
Even I am facing the same error and I have got a warning msg as well
Warning messages:
1: Unnamed `col_types` should have the same length as `col_names`. Using smaller of the two. 
2: In rbind(names(probs), probs_f) :
  number of columns of result is not a multiple of vector length (arg 1)


I am trying to download the BRCA data
my query

query <- GDCquery(project = "TCGA-BRCA", 
                  data.category = "Gene expression",
                  data.type = "Gene expression quantification",
                  experimental.strategy = "RNA-Seq",
                  platform = "Illumina HiSeq",
                  file.type = "results",
                  barcode = samples, 
                  legacy = TRUE)

**Comment by IL2liu** [comment]
I am facing the same error too, the warning message is as following
 |============================================================================| 100%
Downloading genome information (try:0) Using: Homo sapiens genes (GRCh37.p13)
Loading from disk
Starting to add information to samples
 => Add clinical information to samples
Error: lexical error: invalid char in json text.
                                       <?xml version="1.0" ?> <respons
                     (right here) ------^

**Comment by algunion** [comment]
@DarioS,

Please try the following code:

`RNAquery <- GDCquery(project = "TCGA-SKCM",  data.category = "Transcriptome Profiling",
                     data.type = "Gene Expression Quantification",
                     workflow.type = "HTSeq - Counts")
GDCdownload(RNAquery, files.per.chunk = 4)
RNAcounts <- GDCprepare(RNAquery)`

Your argument `chunks.per.download` is invalid.

**Comment by DarioS** [comment]
No, that parameter name change happened recently. It's not the cause of the problem.

**Comment by algunion** [comment]
@DarioS, 

Please execute the code I provided. It should work if you have the last TCGAbiolinks version installed.

**Comment by DarioS** [comment]
Note that R does parameter name checking before functions are executed.

```
> GDCdownload(RNAquery, myBirthday = 4) # 4 April
Error in GDCdownload(RNAquery, myBirthday = 4) : 
  unused argument (myBirthday = 4)
```

Notice that the error message is not about XML. The parameter you mention was changed only two weeks ago. See https://github.com/BioinformaticsFMRP/TCGAbiolinks/issues/139 I think you are conflating two different issues. I guess that it's confusing because the developers made the parameter name change to the release and development branches, which is not supposed to happen according to Bioconductor policies (changes breaking backwards compatibility should not be merged with the release branch).

**Comment by algunion** [comment]
@DarioS,

Note that using three-dot-construct (`...`) in function definition enables the input of extra-arguments without raising the "unused argument" error.

Please use the latest TCGABiolinks version and see if the initial error is still there.

P. S. I am simply a user of this package and trying to help. If you need to report some Bioconductor policy breaking, maybe this is not the right place.

**Comment by DarioS** [comment]
Thanks for volunteering your suggestions. More participation is better. The ellipsis (three-dot) in R does allow inputs of non-specific arguments. However, note that the function prototype of `GDCdownload` does have ellipsis.

```
GDCdownload(query, token.file, method = "api", directory = "GDCdata", files.per.chunk = NULL)
```

In contrast, other R functions (e.g. `rbind`) do have ellipsis in their definition, so can accept non-specified parameters.

Anyway, I upgraded to version 2.5.9 and the issue no longer happens. It would be good to get a description of how the problem was solved by one of the developers, so that we learn something new.
